core/templates/hash_map.h

```diff
@@ -353,6 +353,38 @@ class HashMap {
 		return true;
 	}
 
+	bool replace_key(const TKey &p_old_key, const TKey &p_new_key) {
+		uint32_t pos = 0;
+		bool exists = _lookup_pos(p_old_key, pos);
```

Comment:
Should it be checked, if the new key exists?
```c++
if (_lookup_pos(p_new_key, pos)) { return; }
```

Reply:
Was happening 3 lines down, but I moved the elements access below it.

Reply:
Please note, that I talk about `p_new_key` and not `p_old_key`.

Reply:
Reading is hard. I added that check and also an early return for if the old and new key are equal.